ci_framework/roles/edpm_deploy_baremetal/tasks/main.yml

```diff
@@ -115,6 +115,63 @@
   ansible.builtin.debug:
     var: compute_nodes_output.stdout_lines
 
+- name: Patch OpenStackDataPlane to add repo-setup-downstream service
```

Comment:
Just a couple of suggestions:

1. If we can add the service before applying the CR will be nice, not so sure that for a this service it will make any real difference, but, just an idea.
2. We should create a task for tracking the dev-preview things, so when it's possible come bakc here and  remove it. Here you have an example for a similar situation. https://issues.redhat.com/browse/OSPCIX-58

Reply:
@pablintino 

Thanks for the suggestion.

1>> If we can add the service before applying the CR will be nice, not so sure that for a this service it will make any real difference, but, just an idea.

Based on my understanding, since the deployStrategy is set to false by default in the openstackdataplane CR, when we creating the service should not make any difference as long as it's done before we change the deployStrategy to false, which we do on line 175 in this file. The dev-preview job is green with this change, so I believe we are in good shape.

2>> We should create a task for tracking the dev-preview things, so when it's possible come bakc here and remove it. Here you have an example for a similar situation. https://issues.redhat.com/browse/OSPCIX-58

While there are inline code comments (regarding the to-do of removing these extra logins from dev-preview#1), I agree with your suggestion that creating a Jira card would be beneficial for tracking. I've gone ahead and created a card (OSP-28082) under our dev-preview#2 story to eliminate the dev-preview#1 workaround once we remove that line.

Reply:
Thanks for considering the suggestions, I'll approve it